// @audiotool/nexus/dist/api.js
var d = Object.defineProperty;
var U = (e, n, s) => n in e ? d(e, n, { enumerable: !0, configurable: !0, writable: !0, value: s }) : e[n] = s;
var a = (e, n, s) => U(e, typeof n != "symbol" ? n + "" : n, s);
import { P as E, S as m } from "./preset_service_connect-CdqVX87_.js";
import { az as P, aF as A, a0 as q, a1 as c, C as g, al as j, am as O, O as G, Q as F, a8 as v, a9 as L, o as f, q as K, ap as Y, aq as H, X as x, Y as B, aa as w, ab as J, y as V, z as h, D as M, c as b, av as z, au as $, aw as k, ax as y, a4 as Q, a5 as W, aj as X, ak as Z, M as ee, N as se, u as ae, v as te, e as oe, f as ne, G as re, b as ie, ay as Re, ah as pe, ai as me, a6 as le, a7 as ue, J as Ee, K as Ne, m as Ce, n as de, a2 as Ue, a3 as Se, L as Te, a as _e, A as De, B as Ie, _ as Pe, $ as Ae, at as qe, ar as ce, aE as ge, as as je, af as Oe, aD as Ge, ae as Fe, aC as ve, H as Le, E as fe, F as Ke, aB as Ye, p as He, I as xe, l as Be, Z as we, ag as Je, an as Ve, ao as he, V as Me, W as be, ac as ze, ad as $e, w as ke, x as ye, g as Qe, h as We, U as Xe, d as Ze, i as es, j as ss, R as as, T as ts, r as os, t as ns, k as rs, aA as is, s as Rs } from "./preset_service_connect-CdqVX87_.js";
import { Message as N, proto3 as o, Timestamp as C } from "@bufbuild/protobuf";
import { B as ms, f as ls, E as us, F as Es, d as Ns, e as Cs, I as ds, b as Us, L as Ss, P as Ts, a as _s, S as Ds, T as Is, U as Ps, c as As, g as qs, j as cs, m as gs, k as js, h as Os, i as Gs, l as Fs, p as vs, s as Ls, u as fs } from "./opt_pb-Dp9pF04Q.js";
var R = /* @__PURE__ */ ((e) => (e[e.SYNC_TRACK_SCREENSHOT_ENUM_UNSPECIFIED = 0] = "SYNC_TRACK_SCREENSHOT_ENUM_UNSPECIFIED", e[e.SYNC_TRACK_SCREENSHOT_RUNNING = 1] = "SYNC_TRACK_SCREENSHOT_RUNNING", e[e.SYNC_TRACK_SCREENSHOT_FINISHED = 2] = "SYNC_TRACK_SCREENSHOT_FINISHED", e))(R || {});
o.util.setEnumType(R, "audiotool.project.v1.SyncTrackScreenshotStatus", [
  { no: 0, name: "SYNC_TRACK_SCREENSHOT_ENUM_UNSPECIFIED" },
  { no: 1, name: "SYNC_TRACK_SCREENSHOT_RUNNING" },
  { no: 2, name: "SYNC_TRACK_SCREENSHOT_FINISHED" }
]);
const r = class r extends N {
  constructor(s) {
    super();
    /**
     * The project that is synced.
     *
     * @generated from field: audiotool.project.v1.Project project = 1;
     */
    a(this, "project");
    /**
     * The commit index of the project.
     *
     * @generated from field: uint32 commit_index = 2;
     */
    a(this, "commitIndex", 0);
    /**
     * The desired track name.
     *
     * @generated from field: string track_name = 3;
     */
    a(this, "trackName", "");
    /**
     * The user who triggered the sync.
     *
     * @generated from field: string user_name = 4;
     */
    a(this, "userName", "");
    /**
     * The time when the operation was created.
     *
     * @generated from field: google.protobuf.Timestamp create_time = 5;
     */
    a(this, "createTime");
    /**
     * The current status of the sync track process.
     *
     * @generated from field: audiotool.project.v1.SyncTrackScreenshotStatus status = 6;
     */
    a(this, "status", 0);
    o.util.initPartial(s, this);
  }
  static fromBinary(s, t) {
    return new r().fromBinary(s, t);
  }
  static fromJson(s, t) {
    return new r().fromJson(s, t);
  }
  static fromJsonString(s, t) {
    return new r().fromJsonString(s, t);
  }
  static equals(s, t) {
    return o.util.equals(r, s, t);
  }
};
a(r, "runtime", o), a(r, "typeName", "audiotool.project.v1.SyncTrackScreenshotInfo"), a(r, "fields", o.util.newFieldList(() => [
  { no: 1, name: "project", kind: "message", T: E },
  {
    no: 2,
    name: "commit_index",
    kind: "scalar",
    T: 13
    /* ScalarType.UINT32 */
  },
  {
    no: 3,
    name: "track_name",
    kind: "scalar",
    T: 9
    /* ScalarType.STRING */
  },
  {
    no: 4,
    name: "user_name",
    kind: "scalar",
    T: 9
    /* ScalarType.STRING */
  },
  { no: 5, name: "create_time", kind: "message", T: C },
  { no: 6, name: "status", kind: "enum", T: o.getEnumType(R) }
]));
let l = r;
var p = /* @__PURE__ */ ((e) => (e[e.UNSPECIFIED = 0] = "UNSPECIFIED", e[e.STARTED = 1] = "STARTED", e[e.AUDIO_RENDERING = 3] = "AUDIO_RENDERING", e[e.AUDIO_RENDERED = 4] = "AUDIO_RENDERED", e[e.AUDIO_CONVERTING = 5] = "AUDIO_CONVERTING", e[e.AUDIO_CONVERTED = 6] = "AUDIO_CONVERTED", e[e.FINISHED = 7] = "FINISHED", e))(p || {});
o.util.setEnumType(p, "audiotool.project.v1.SyncTrackStatus", [
  { no: 0, name: "SYNC_TRACK_STATUS_UNSPECIFIED" },
  { no: 1, name: "SYNC_TRACK_STATUS_STARTED" },
  { no: 3, name: "SYNC_TRACK_STATUS_AUDIO_RENDERING" },
  { no: 4, name: "SYNC_TRACK_STATUS_AUDIO_RENDERED" },
  { no: 5, name: "SYNC_TRACK_STATUS_AUDIO_CONVERTING" },
  { no: 6, name: "SYNC_TRACK_STATUS_AUDIO_CONVERTED" },
  { no: 7, name: "SYNC_TRACK_STATUS_FINISHED" }
]);
const i = class i extends N {
  constructor(s) {
    super();
    /**
     * The project that is synced.
     *
     * @generated from field: audiotool.project.v1.Project project = 1;
     */
    a(this, "project");
    /**
     * Required. The mode to sync the track.
     *
     * @generated from field: audiotool.project.v1.SyncTrackMode mode = 2;
     */
    a(this, "mode", m.UNSPECIFIED);
    /**
     * The commit index of the project.
     *
     * @generated from field: uint32 commit_index = 3;
     */
    a(this, "commitIndex", 0);
    /**
     * The desired track name.
     *
     * @generated from field: string track_name = 4;
     */
    a(this, "trackName", "");
    /**
     * The user who triggered the sync.
     *
     * @generated from field: string user_name = 5;
     */
    a(this, "userName", "");
    /**
     * The time when the operation was created.
     *
     * @generated from field: google.protobuf.Timestamp create_time = 6;
     */
    a(this, "createTime");
    /**
     * The current status of the sync track process.
     *
     * @generated from field: audiotool.project.v1.SyncTrackStatus status = 7;
     */
    a(this, "status", 0);
    o.util.initPartial(s, this);
  }
  static fromBinary(s, t) {
    return new i().fromBinary(s, t);
  }
  static fromJson(s, t) {
    return new i().fromJson(s, t);
  }
  static fromJsonString(s, t) {
    return new i().fromJsonString(s, t);
  }
  static equals(s, t) {
    return o.util.equals(i, s, t);
  }
};
a(i, "runtime", o), a(i, "typeName", "audiotool.project.v1.SyncTrackInfo"), a(i, "fields", o.util.newFieldList(() => [
  { no: 1, name: "project", kind: "message", T: E },
  { no: 2, name: "mode", kind: "enum", T: o.getEnumType(m) },
  {
    no: 3,
    name: "commit_index",
    kind: "scalar",
    T: 13
    /* ScalarType.UINT32 */
  },
  {
    no: 4,
    name: "track_name",
    kind: "scalar",
    T: 9
    /* ScalarType.STRING */
  },
  {
    no: 5,
    name: "user_name",
    kind: "scalar",
    T: 9
    /* ScalarType.STRING */
  },
  { no: 6, name: "create_time", kind: "message", T: C },
  { no: 7, name: "status", kind: "enum", T: o.getEnumType(p) }
]));
let u = i;
export {
  P as Audiograph,
  A as AudiographService,
  ms as Bool,
  ls as Bytes,
  q as CloseSessionRequest,
  c as CloseSessionResponse,
  g as CommentMode,
  j as CreatePresetRequest,
  O as CreatePresetResponse,
  G as CreateProjectRequest,
  F as CreateProjectResponse,
  v as CreateProjectRoleRequest,
  L as CreateProjectRoleResponse,
  f as CreateSampleRequest,
  K as CreateSampleResponse,
  Y as DeletePresetRequest,
  H as DeletePresetResponse,
  x as DeleteProjectRequest,
  B as DeleteProjectResponse,
  w as DeleteProjectRoleRequest,
  J as DeleteProjectRoleResponse,
  V as DeleteSampleRequest,
  h as DeleteSampleResponse,
  M as DeleteUserRequest,
  b as DeleteUserResponse,
  us as EntityOptions,
  Es as FieldOptions,
  Ns as Float,
  Cs as FloatRange,
  z as GetAudiographChannels,
  $ as GetAudiographResolution,
  k as GetAudiographsRequest,
  y as GetAudiographsResponse,
  Q as GetLatestVersionBundleRequest,
  W as GetLatestVersionBundleResponse,
  X as GetPresetRequest,
  Z as GetPresetResponse,
  ee as GetProjectRequest,
  se as GetProjectResponse,
  ae as GetSampleRequest,
  te as GetSampleResponse,
  oe as GetSettingsRequest,
  ne as GetSettingsResponse,
  re as GetUserRequest,
  ie as GetUserResponse,
  Re as Graph,
  ds as Int32,
  Us as Int32Range,
  Ss as ListOptions,
  pe as ListPresetsRequest,
  me as ListPresetsResponse,
  le as ListProjectRolesRequest,
  ue as ListProjectRolesResponse,
  Ee as ListProjectsRequest,
  Ne as ListProjectsResponse,
  Ce as ListSamplesRequest,
  de as ListSamplesResponse,
  Ue as ListSessionsRequest,
  Se as ListSessionsResponse,
  Te as ListUsersRequest,
  _e as ListUsersResponse,
  De as ListenRequest,
  Ie as ListenResponse,
  Pe as OpenSessionRequest,
  Ae as OpenSessionResponse,
  Ts as PointerOptions,
  qe as Preset,
  ce as PresetDeviceType,
  ge as PresetService,
  je as PresetUsage,
  E as Project,
  Oe as ProjectRole,
  Ge as ProjectRoleService,
  Fe as ProjectRoleType,
  ve as ProjectService,
  Le as SampleConvertDone,
  fe as SampleConvertDoneErrorType,
  Ke as SampleEvent,
  Ye as SampleService,
  He as SampleUploadEndpoint,
  xe as Session,
  Be as Settings,
  _s as String,
  Ds as StringFormat,
  u as SyncTrackInfo,
  m as SyncTrackMode,
  we as SyncTrackRequest,
  l as SyncTrackScreenshotInfo,
  R as SyncTrackScreenshotStatus,
  p as SyncTrackStatus,
  Is as TargetType,
  Je as TrackLicense,
  Ps as UInt32,
  As as UInt32Range,
  Ve as UpdatePresetRequest,
  he as UpdatePresetResponse,
  Me as UpdateProjectRequest,
  be as UpdateProjectResponse,
  ze as UpdateProjectRoleRequest,
  $e as UpdateProjectRoleResponse,
  ke as UpdateSampleRequest,
  ye as UpdateSampleResponse,
  Qe as UpdateSettingsRequest,
  We as UpdateSettingsResponse,
  Xe as UpdateUserRequest,
  Ze as UpdateUserResponse,
  es as UploadAvatarRequest,
  ss as UploadAvatarResponse,
  as as UploadCoverRequest,
  ts as UploadCoverResponse,
  os as UploadSampleFinishedRequest,
  ns as UploadSampleFinishedResponse,
  rs as User,
  is as UserService,
  qs as bool,
  cs as bytes,
  gs as entity,
  js as field,
  Os as float,
  Gs as int32,
  Fs as list,
  vs as pointer,
  Rs as sample,
  Ls as string$,
  fs as uint32
};
